sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -197,55 +193,65 @@ class QueryExecution(
   }
 
   def simpleString: String = withRedaction {
-    s"""== Physical Plan ==
-       |${stringOrError(executedPlan.treeString(verbose = false))}
-      """.stripMargin.trim
+    val rope = new StringRope()
+
+    rope.append("== Physical Plan ==\n")
+    appendOrError(rope.append)(
+      executedPlan.treeString(_, false, false, SQLConf.get.maxToStringFields))
+    rope.append("\n")
+
+    rope.toString
   }
 
-  private def writeOrError(writer: Writer)(f: Writer => Unit): Unit = {
-    try f(writer)
+  private def appendOrError(append: String => Unit)(f: (String => Unit) => Unit): Unit = {
```

Comment:
You always call this method on a QueryPlan. Can we specialize this method for this scenario and pass the plan and all the needed treeString parameters?

Different question. Is it possible that treeString already writes to the appender before throwing an exception. It it does the output might look pretty weird, because it will and contain a part of the tree and the exception thrown.

Reply:
> Can we specialize this method for this scenario and pass the plan and all the needed treeString parameters?

@hvanhovell Do you mean changes like in the PR https://github.com/MaxGekk/spark-1/pull/16 ? Unfortunately it doesn't work well because plan's constructor can produce `AnalysisException` which cannot be handled with this approach.

Reply:
Yeah that would be it. You can pass in a call-by-name parameter if you want to capture errors during construction. I prefer his over having some overly generic function.

Reply:
Will be ok if I move `appendOrError` to the companion object of `QueryPlan`? Like:
```scala
object QueryPlan extends PredicateHelper {
  /**
   * Converts the query plan to string and appends it via provided function.
   */
  def append[T <: QueryPlan[T]](
      plan: => QueryPlan[T],
      append: String => Unit,
      verbose: Boolean,
      addSuffix: Boolean,
      maxFields: Int = SQLConf.get.maxToStringFields): Unit = { () }
```

Reply:
Sure that works.